### PR TITLE
Initial support for OCI package lifecycle

### PR DESCRIPTION
**Pull request by justinsb**
We store the lifecycle in the kpt.dev/lifecycle annotation.

### CODE REVIEW

File: porch/pkg/oci/mutate.go [hunk 1]
```diff
@@ -201,7 +201,31 @@ func (p *ociPackageDraft) Close(ctx context.Context) (repository.PackageRevision
 
 	klog.Infof("pushing %s", ref)
 
-	img, err := mutate.Append(p.base, p.addendums...)
+	addendums := append([]mutate.Addendum{}, p.addendums...)
+	if p.lifecycle != "" {
+		if len(addendums) == 0 {
```
Comment (mortent) [inline porch/pkg/oci/mutate.go:206]: When do we encounter this situation? Would this be when we approve an oci package, thus we want to update the lifecycle but there isn't any other changes to the content that requires a new layer?

**Review by mortent — APPROVED** [review]
Just a small question for clarification.

Also, should we add some tests for this?

File: porch/pkg/oci/mutate.go (hunk 1 above)
Comment (justinsb) [reply porch/pkg/oci/mutate.go:206]: That's exactly right.  I think it will likely happen whenever we start approving packages.  I figure it's an important edge case that we're going to have to handle, but right now as I don't think we have test coverage I just surface an error.  I'd like to improve the test coverage here e.g. running all tests against both git & oci Repositories, as the API behaviour should be equivalent.
